### PR TITLE
Create separate mega menus for desktop and mobile

### PR DESCRIPTION
**Pull request by Rleahy22**
## Description
department-of-veterans-affairs/va.gov-team#25659
Creates a mega menu for desktop and one for mobile to allow proper tab ordering
This change is in conjunction with changes to
vets-website: https://github.com/department-of-veterans-affairs/vets-website/pull/18116
veteran-facing-services-tools: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/655

## Testing done
local

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/includes/top-nav.html [hunk 1]
```diff
@@ -31,13 +31,20 @@
     </div>
     {% if !noNavOrLogin %}
       <div id="va-nav-controls"></div>
+      {% if !noMegamenu %}
```
Comment (mpelzsherman) [inline src/site/includes/top-nav.html:34]: The double negative here (`!` + `no`) is confusing. Could we change it to `if megaMenu`?

**Comment by mpelzsherman** [comment]
I'd like to test this in your review environment, but you have a failing test that's killing the build.

**Comment by mpelzsherman** [comment]
Tested locally, and it looks like the subnav items are missing?

Locations___Nashville_Vet_Center___Veterans_Affairs

Locations___Nashville_Vet_Center___Veterans_Affairs

**Comment by Rleahy22** [comment]
> Tested locally, and it looks like the subnav items are missing?
> 
> Locations___Nashville_Vet_Center___Veterans_Affairs
> 
> Locations___Nashville_Vet_Center___Veterans_Affairs

Right, this change relies on simultaneous changes from vets-website, and veteran-facing-service-tools. I'm not sure the best way to go about merging them or how to have them tested as each repo needs the changes from the other. What would you suggest I do in this situation, @mpelzsherman ?

**Comment by mpelzsherman** [comment]
> This change relies on simultaneous changes from vets-website, and veteran-facing-service-tools. I'm not sure the best way to go about merging them or how to have them tested as each repo needs the changes from the other. What would you suggest I do in this situation, @mpelzsherman ?

Can we put a feature flip around these changes? That way we can wait until all of the changes have been merged and deployed, then flip the feature on in staging to test, then prod if it passes. Once everything is working, the feature flip code can be removed.

@timwright12 - any other suggestions?

File: src/site/includes/top-nav.html (hunk 1 above)
Comment (Rleahy22) [reply src/site/includes/top-nav.html:34]: I believe `noMegaMenu` comes from the CMS, `if megaMenu` would always return false.

Comment (mpelzsherman) [reply src/site/includes/top-nav.html:34]: Are you sure? I see it [here](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/applications/registry.json#L334), but I do not see it in the Drupal debug content. Of course I could always be wrong!

Comment (Rleahy22) [reply src/site/includes/top-nav.html:34]: Definitely not sure, hence "I believe." I'll update this

**Pull request merged**